main.tf

```
@@ -91,6 +91,17 @@ resource "aws_rds_cluster" "this" {
     }
   }
 
+  dynamic "restore_to_point_in_time" {
+    for_each = length(keys(var.restore_to_point_in_time)) == 0 ? [] : [var.restore_to_point_in_time]
+
+    content {
+      source_cluster_identifier  = lookup(restore_to_point_in_time.value, "source_cluster_identifier", null)
```

this attribute is required and should be:
```suggestion
      source_cluster_identifier  = restore_to_point_in_time.value.source_cluster_identifier
```